### ISSUE TITLE
[BUG/Help] p-tuning实现是否有误？

### ISSUE THREAD

**Issue by dimfractal** [THUDM/ChatGLM-6B#519]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

![image](https://user-images.githubusercontent.com/29627984/231099760-06372081-19d9-4de6-a4e6-34da75fcf910.png)
https://github.com/THUDM/ChatGLM-6B/blob/main/ptuning/main.py
p-tuning 的实现是否有错误？221行的mask_position应为context_length？
```python
context_length = input_ids.index(tokenizer.bos_token_id)
mask_position = context_length - 1
labels = [-100] * context_length + input_ids[context_length+1:]
```

### Steps To Reproduce

No

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by duzx16** [comment]
没问题。按你的修改 `input_ids` 和 `labels` 长度就不一样了。right shift 是在 `ChatGLMForConditionalGeneration.forward` 里做的，不需要手动做

**Comment by nghuyong** [comment]
https://github.com/THUDM/ChatGLM-6B/blob/main/ptuning/main.py#L221
按照现在的实现，bos_token_id 就被计算loss了，实际这个token不用计算loss吧

所以应该是：

```
context_length = input_ids.index(tokenizer.bos_token_id) + 1
labels = [-100] * context_length + input_ids[context_length:]
```
